build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/StringTemplatePlugin.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.file.ConfigurableFileTree;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
+import org.gradle.api.tasks.TaskProvider;
+
+import java.io.File;
+
+public class StringTemplatePlugin implements Plugin<Project> {
+    @Override
+    public void apply(Project project) {
+        File outputDir = project.file("src/main/generated-src/");
+
+        TaskProvider<StringTemplateTask> generateSourceTask = project.getTasks().register("stringTemplates", StringTemplateTask.class);
+        generateSourceTask.configure(stringTemplateTask -> stringTemplateTask.getOutputFolder().set(outputDir));
+        project.getPlugins().withType(JavaPlugin.class, javaPlugin -> {
+            SourceSetContainer sourceSets = project.getExtensions().getByType(JavaPluginExtension.class).getSourceSets();
+            SourceSet mainSourceSet = sourceSets.getByName(SourceSet.MAIN_SOURCE_SET_NAME);
+            ConfigurableFileTree outputFileTree = project.fileTree(outputDir);
+            outputFileTree.builtBy(generateSourceTask);
+            mainSourceSet.getJava().srcDir(generateSourceTask);
+            project.getTasks().named(mainSourceSet.getCompileJavaTaskName()).configure(task -> task.dependsOn(generateSourceTask));
```

Comment:
I would think calling `srcDir` above would create an implicit dependency here but perhaps not.

Reply:
Local testing shows that this is indeed not needed. Removed.